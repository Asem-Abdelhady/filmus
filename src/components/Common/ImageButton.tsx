import {
    Button,
    Image,
  } from "@chakra-ui/react";
  import React from "react";
  
  const ImageButton = ({colorScheme='teal', src, width}) => {
    return (
      <Button colorScheme={colorScheme} variant='outline' size='sm'>
        <Image
          width={width}
          src={src}
          alt='btn_image'
        />
      </Button>
    );
  };
  
  export default ImageButton;
  