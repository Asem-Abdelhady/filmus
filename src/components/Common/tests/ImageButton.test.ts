import renderer from 'react-test-renderer';
import ImageButton from '../ImageButton';

it('renders the image button', () => {
  const component = renderer.create(
    <ImageButton src="src" width={100}></ImageButton>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});