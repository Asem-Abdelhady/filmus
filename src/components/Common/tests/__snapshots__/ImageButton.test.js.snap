// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-5c2176{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-appearance:none;-moz-appearance:none;-ms-appearance:none;appearance:none;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;position:relative;white-space:nowrap;vertical-align:middle;outline:2px solid transparent;outline-offset:2px;width:auto;}",
      }
    }
    data-emotion="css 5c2176"
  />,
  <button
    className="chakra-button css-5c2176"
    type="button"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-v7v99c{width:100px;}",
        }
      }
      data-emotion="css v7v99c"
    />
    <img
      alt="btn_image"
      className="chakra-image css-v7v99c"
      src="src"
    />
  </button>,
]
`;
